DataLayer/BatteryData/BatteryData.h

```diff
@@ -9,63 +9,93 @@ class BatteryData : public I_BatteryData
-    void setMod1CellVoltages(QList<double> mod1CellVoltages);
-
-    void setMod2PcbTemperature(double mod2PcbTemperature);
-    void setMod2CellTemperature(double mod2CellTemperature);
-    void setMod2CellVoltages(QList<double> mod2CellVoltages);
+    void setAlive(const unsigned char&);
+    void setPackSocAmpHours(const float&);
+    void setPackSocPercentage(const float&);
+    void setPackBalanceSoc(const float&);
+    void setPackBalanceSocPercentage(const float&);
+    void setChargingCellVoltageError(const unsigned short&);
+    void setCellTemperatureMargin(const unsigned short&);
+    void setDischargingCellVoltageError(const unsigned short&);
+    void setTotalPackCapacity(const unsigned short&);
+    void setPrechargeContactorDriverStatus(const unsigned char&);
+    void setPrechargeState(const unsigned char&);
+    void setPrechargeTimerElapsed(const unsigned char&);
+    void setPrechargeTimerCount(const unsigned short&);
+    void setLowestCellVoltage(const unsigned short&);
+    void setLowestCellVoltageNumber(const unsigned char&);
+    void setHighestCellVoltage(const unsigned short&);
+    void setHighestCellVoltageNumber(const unsigned char&);
+    void setLowestCellTemperature(const unsigned short&);
+    void setLowestCellTemperatureNumber(const unsigned char&);
+    void setHighestCellTemperature(const unsigned short&);
+    void setHighestCellTemperatureNumber(const unsigned char&);
+    void setVoltage(const unsigned int&);
+    void setCurrent(const unsigned int&);
+    void setFan0Speed(const unsigned short&);
+    void setFan1Speed(const unsigned short&);
+    void setFanContactors12VCurrentConsumption(const unsigned short&);
+    void setCmu12VCurrentConsumption(const unsigned short&);
+    void setBmsCanLockedOut(const unsigned char&);
 
-    void setMod3PcbTemperature(double mod3PcbTemperature);
-    void setMod3CellTemperature(double mod3CellTemperature);
-    void setMod3CellVoltages(QList<double> mod3CellVoltages);
-
-    void setBatteryVoltage(double batteryVoltage);
-    void setBatteryCurrent(double batteryCurrent);
 
 private:
-    double mod0PcbTemperature_;
-    double mod0CellTemperature_;
-    QList<double> mod0CellVoltages_;
-
-    double mod1PcbTemperature_;
-    double mod1CellTemperature_;
-    QList<double> mod1CellVoltages_;
-
-    double mod2PcbTemperature_;
-    double mod2CellTemperature_;
-    QList<double> mod2CellVoltages_;
-
-    double mod3PcbTemperature_;
-    double mod3CellTemperature_;
-    QList<double> mod3CellVoltages_;
-
-    double batteryVoltage_;
-    double batteryCurrent_;
+    unsigned char alive_;
+    float packSocAmpHours_;
+    float packSocPercentage_;
+    float packBalanceSoc_;
+    float packBalanceSocPercentage_;
+    unsigned short chargingCellVoltageError_;
+    unsigned short cellTemperatureMargin_;
+    unsigned short dischargingCellVoltageError_;
+    unsigned short totalPackCapacity_;
+    unsigned char prechargeContactorDriverStatus_;
```

Comment:
Once it gets to the data layer, the info should be as nice as possible, there shouldn't be any need to deal with bit fields. So stuff like this should be split up at this point to.

```
Contactor0ErrorStatus
Contactor1ErrrorStatus
...
Contactor2Status
```